scheds/rust/scx_lavd/src/bpf/lock.bpf.c

```diff
@@ -7,46 +7,60 @@
 /*
  * To be included to the main.bpf.c
  */
-
-static void inc_futex_boost(u32 *uaddr)
+static void __inc_futex_boost(struct cpu_ctx *cpuc)
 {
 	struct task_struct *p = bpf_get_current_task_btf();
 	struct task_ctx *taskc = try_get_task_ctx(p);
-	struct cpu_ctx *cpuc = get_cpu_ctx();
 
-	if (taskc && cpuc && (taskc->futex_uaddr != uaddr)) {
-		taskc->futex_boost++;
-		taskc->futex_uaddr = uaddr;
-		cpuc->lock_holder = is_lock_holder(taskc);
+	if (taskc) {
+		if (!cpuc)
+			cpuc = get_cpu_ctx();
+
+		if (cpuc) {
+			taskc->futex_boost = true;
+			cpuc->lock_holder = true;
+		}
 	}
 	/*
 	 * If taskc is null, the task is not under sched_ext so ignore the error.
 	 */
 }
 
-static void dec_futex_boost(u32 *uaddr)
+static void __dec_futex_boost(struct cpu_ctx *cpuc)
 {
 	struct task_struct *p = bpf_get_current_task_btf();
 	struct task_ctx *taskc = try_get_task_ctx(p);
-	struct cpu_ctx *cpuc = get_cpu_ctx();
 
-	if (p && taskc && cpuc && taskc->futex_boost > 0) {
-		taskc->futex_boost--;
-		taskc->futex_uaddr = NULL;
-		cpuc->lock_holder = is_lock_holder(taskc);
+	if (taskc && taskc->futex_boost) {
+		if (!cpuc)
```

Comment:
nit: you might be able to push this lookup into `dec_futex_boost` and that keeps this function a little cleaner where it can return early if `!cpuc`.